### ISSUE TITLE
508-defect-2 [SCREENREADER, COGNITION]: Images of the facilities should not be links

### ISSUE THREAD

**Issue by joshkimux** [department-of-veterans-affairs/va.gov-team#25023]
# [508-defect-2](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/accessibility/guidance/defect-severity-rubric.md#508-defect-2)

## Feedback framework

- **❗️ Must** for if the feedback must be applied 
- **⚠️ Should** if the feedback is best practice 
- **✔️ Consider** for suggestions/enhancements 

## Definition of done

1. Review and acknowledge feedback.
1. Fix and/or document decisions made.
1. Accessibility specialist will close ticket after reviewing documented decisions / validating fix.

## Point of Contact

**VFS Point of Contact:** Josh

## User Story or Problem Statement

As a screen reader user, I want to know the purpose and destination of a link.
As a keyboard user, I want to know why an image is receiving focus and where it will direct me to.

## Details

Images on the locations page all contain blank links. For screen reader users, that means alt text describing the facilities will populate their screenreader's link list which can be disorientating as alt text describing a facility will not describe where they will be linked to. For keyboard users, that means trying to figure out _why_ an image is linked to begin with. Without any contextual information around it, they will not understand the purpose or target.

## Acceptance Criteria

- [x] Images of facilities are not links

## Proposed Solution

If there's no particular intent behind the links on the images, they should be removed.

## Screenshots or Trace Logs

https://user-images.githubusercontent.com/14154792/119010093-b552c980-b961-11eb-96d6-13b67988264f.mov

**Comment by mmiddaugh** [comment]
Additional info noted in duplicate issue: Redundant links on Vet Center home page #24749

**Defect**
Adjacent links (within the photo and map) go to the same URL.
Why It Matters: this results in additional navigation and repetition for keyboard and screen reader users

**Code**

```
<a id="generated-mapbox-image-link" href="https://maps.google.com?saddr=Current+Location&daddr=3500 Ludington Street, Suite 110, Escanaba, MI 49829" target="_blank" rel="noopener noreferrer">
<div class="va-c-position--relative vads-u-display--inline-block">
<span class="vads-u-margin-right--1p5 vads-u-margin-top--1p5 vads-u-text-decoration--none vads-u-display--flex vads-u-align-items--center expand-image-button va-c-position--absolute va-c-position-top-right-corner vads-u-justify-content--center">
<i class="fas fa-expand-arrows-alt"></i>
</span>
```
---
## Tasks
- [ ] First step: remove redundant link
- [ ] Add listener to track clicks on the image
- [ ] Map link should be maintained from the map thumbnail 

## Acceptance Criteria
- [ ] Adjacent links will be discernable
- [ ] The photo is not clickable
- [ ] Clicks on the photo are tracked
- [ ] Veteran can view the map in a new tab by clicking on the map

**Comment by mpelzsherman** [comment]
Suggestion: let's avoid writing tasks with implementation-specific solutions. e.g. "Add listener to track clicks on the image". It's usually better to describe the requirements, and let the developers find the best solution.

**Comment by mpelzsherman** [comment]
@mmiddaugh So just to be clear: we want to track clicks on the photo just to see how often people click on them, right?
What tracking parameters should we use for this?

**Comment by mpelzsherman** [comment]
@mmiddaugh I have left the tracking parameters on the photo the same as they are currently; hope that's OK.

@timwright12 raised a good point in my PR: photo "clicks" won't trigger for keyboard users since removing the link makes it non-focusable. Is that OK? I'm not sure how to make the photo focusable without wrapping it in a link, or if that's even desirable from an a11y standpoint.

**Comment by joshkimux** [comment]
@mpelzsherman That's OK as keyboard users have access to "Get directions" which provides the same action 👍

**Comment by mpelzsherman** [comment]
A tugboat review environment is now available: https://web-eyv4neniupjueqkjymmjbfcjvbxvtxkr.demo.cms.va.gov/escanaba-vet-center/

Note: you will need to disable CORS in your browser, LMK if you need any help with that.

**Comment by mpelzsherman** [comment]
I went ahead and merged my PR because I'm pretty sure the changes are OK, and I did some refactoring that we'd like to take advantage of. There's no risk of impact to production since the Vet Centers haven't been published in Drupal yet.

**Comment by mmiddaugh** [comment]
We need to do some fine tuning on this one - the photos appear to be actionable, creating an expectation that something will happen when the user clicks on the image. Will groom next week.

**Comment by mmiddaugh** [comment]
Change needs to be added to Locations page (not just on Main)

**Comment by joshkimux** [comment]
For the most part, this is working. Images are no longer appearing in the links list, but it looks like the map can still receive focus as part of the natural tab order.

https://user-images.githubusercontent.com/14154792/123647412-0cb44700-d7f6-11eb-9a60-52a2230e3564.mov

https://user-images.githubusercontent.com/14154792/123647859-746a9200-d7f6-11eb-8165-1db0d4c080df.mov

**Comment by mpelzsherman** [comment]
The map is supposed to be clickable per AC. Is there still a problem here?

**Comment by joshkimux** [comment]
I can clarify a bit more 👍 

As you can see in the second video, the map can receive focus in the tab order but it's not announced to screen reader users. That creates kind of a "ghost" tab stop: they might hit `tab` after "View more ___ locations" and then hear no response.

**Comment by mmiddaugh** [comment]
This can now be validated in Prod 🎉 
https://www.va.gov/nashville-vet-center/
https://www.va.gov/colorado-springs-vet-center/
@AngelaFowler82 @joshkimux

**Comment by AngelaFowler82** [comment]
The blank links are officially gone. :)

**Comment by joshkimux** [comment]
@mmiddaugh Heads up, the issue regarding the ghost tab stop still exists for the image of the map, but that might be best addressed in a separate ticket.
